File: android/src/main/java/ru/lab332/caplugin/ostimer/OsTimerPlugin.java
package ru.lab332.caplugin.ostimer;

import android.os.Handler;
import android.os.Message;
import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "OsTimer")
public class OsTimerPlugin extends Plugin {
    public static String EVENT_NAME = "OsTimerTick";

    private OsTimer osTimer = new OsTimer();


    @PluginMethod(returnType = PluginMethod.RETURN_NONE)
    public void start(PluginCall call) {
        int delay = call.getInt("delay", getConfig().getInt("delay", 1500));
        int period = call.getInt("period", getConfig().getInt("period", 1500));
        osTimer.start(
                (OsTimerCallback) tick -> {
                    JSObject payload = new JSObject();
                    payload.put("tick", tick);
                    notifyListeners(EVENT_NAME, payload);
                },
                delay,
                period
        );

        call.resolve();
    }

    @PluginMethod(returnType = PluginMethod.RETURN_NONE)
    public void stop(PluginCall call) {
        osTimer.stop();
        call.resolve();
    }
}

File: ios/Plugin/OsTimerPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(OsTimerPlugin)
public class OsTimerPlugin: CAPPlugin {
    private var osTimer = OsTimer(timeInterval: 2)
    private var tmrCounter=0
    private static let EVENT_NAME="OsTimerTick"
    
    @objc public func start(_ call: CAPPluginCall){
        /*let delay=call.getString("delay") ?? (
        getConfigValue("delay") as? String ?? "1500")*/
        let period=call.getInt("period") ?? Int(getConfigValue("period") as? String ?? "2000")
         
        osTimer.timeInterval=TimeInterval(period!/1000)
        
        osTimer.eventHandler={
    
            self.tmrCounter=self.tmrCounter+1
            self.notifyListeners(OsTimerPlugin.EVENT_NAME,  data: ["tick":self.tmrCounter])
        }
 
        osTimer.resume()
        call.resolve()
    }

    @objc public func stop(_ call: CAPPluginCall){
        call.resolve()
    }
}

